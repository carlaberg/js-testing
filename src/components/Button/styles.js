import styled from 'styled-components'

export const StyledButton = styled.button`
  background: deeppink;
  color: white;
  border: none;
  padding: 10px 20px;
  border-radius: 8px;
`