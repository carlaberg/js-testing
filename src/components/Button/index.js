import React from 'react'
import PropTypes from 'prop-types'
import { StyledButton } from './styles'

const Button = ({ text }) => <StyledButton type="button">{text}</StyledButton>

export default Button

Button.propTypes = {
  text: PropTypes.string
}

Button.defaultProps = {
  text: 'I am a button'
}
