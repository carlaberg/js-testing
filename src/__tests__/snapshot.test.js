import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow, mount } from 'enzyme'
import { StyledButton } from '../components/Button/styles'
import Button from '../components/Button'

// Jest styled components to help facilitate testing of component styling
// when using css in JS library styled components
import 'jest-styled-components'

describe('Button', () => {

  it('React component matches snapshot', () => {

    // Import React component to test features that is not related to styling
    const reactComponent = mount(<Button />)
    expect(toJson(reactComponent)).toMatchSnapshot()
  })

  // Import styled component to test features that is related to styling
  it('Styled component matches snapshot', () => {
    const styles = shallow(<StyledButton />)
    expect(toJson(styles)).toMatchSnapshot()
  })
})