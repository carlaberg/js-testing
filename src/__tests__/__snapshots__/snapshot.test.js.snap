// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button React component matches snapshot 1`] = `
.c0 {
  background: deeppink;
  color: white;
  border: none;
  padding: 10px 20px;
  border-radius: 8px;
}

<Button
  text="I am a button"
>
  <styles__StyledButton
    type="button"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "styles__StyledButton-qg1tyq-0",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "background:deeppink;color:white;border:none;padding:10px 20px;border-radius:8px;",
            ],
          },
          "displayName": "styles__StyledButton",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "styles__StyledButton-qg1tyq-0",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      type="button"
    >
      <button
        className="c0"
        type="button"
      >
        I am a button
      </button>
    </StyledComponent>
  </styles__StyledButton>
</Button>
`;

exports[`Button Styled component matches snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "styles__StyledButton-qg1tyq-0",
        "isStatic": false,
        "lastClassName": "jqvfhR",
        "rules": Array [
          "background:deeppink;color:white;border:none;padding:10px 20px;border-radius:8px;",
        ],
      },
      "displayName": "styles__StyledButton",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "styles__StyledButton-qg1tyq-0",
      "target": "button",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
